### PR TITLE
Support multiple users and mapping for localvm

### PR DESCRIPTION
**Pull request by srieger1**
added support for multiple users and mapping users to VMs. Can be used as a solution to select instances based on userid. Also offering reliable selection of VM when testing multiple users locally. With availableInstancesList.pop(), VMs as instances could only be consumed once and backend needs to be restarted afterwards.

added notification for user feedback when deployment fails, e.g., due to improper mapping etc., maybe it would also be good to have notifications for deploy of edited files and reload of environment during assignment. Will open separate PR for that.

Fixed error handling of environment creation. Backend API gave 200 OK also when errors where caught.

### CODE REVIEW

File: backend/src/providers/LocalVMProvider.ts [hunk 1]
```diff
@@ -41,9 +41,40 @@ export default class LocalVMProvider implements InstanceProvider {
     return this.availableInstances.get(identifier);
   }
 
-  async createServer(): Promise<VMEndpoint> {
+  async createServer(identifier: string): Promise<VMEndpoint> {
     if (this.availableInstancesList.length > 0) {
-      return this.availableInstances.get(this.availableInstancesList.pop());
```
Comment (PBug90) [inline backend/src/providers/LocalVMProvider.ts:46]: I suggest we extract the new capability into a new Provider. This preserves the simple logic in this one and allows injecting the more complex one on demand as a dependency.

File: backend/src/authentication/PlaintextAuthenticationProvider.ts [hunk 2]
```diff
@@ -9,8 +9,24 @@ export default class PlaintextAuthenticationProvider
     username: string,
     password: string
   ): Promise<AuthenticationResult> {
-    if (password === "p4") {
-      return { username: username, userid: username, type: "plain" };
+    if (process.env.BACKEND_USERS != undefined) {
```
Comment (PBug90) [inline backend/src/authentication/PlaintextAuthenticationProvider.ts:12]: I suggest we create a new AuthenticationProvider here for the time being so we can handle both use cases easily.

File: frontend/src/views/AssignmentOverview.tsx [hunk 3]
```diff
@@ -7,6 +7,14 @@ import { ListItemSecondaryAction } from '@material-ui/core';
 import List from '@material-ui/core/List';
 import ListItem from '@material-ui/core/ListItem';
 import ListItemText from '@material-ui/core/ListItemText';
+import Snackbar from '@material-ui/core/Snackbar';
```
Comment (PBug90) [inline frontend/src/views/AssignmentOverview.tsx:10]: Thanks for all the frontend changes to make it more usable! :)

**Review by PBug90 — CHANGES_REQUESTED** [review]

File: backend/src/providers/LocalVMProvider.ts (hunk 1 above)
Comment (srieger1) [reply backend/src/providers/LocalVMProvider.ts:46]: good point! I pushed a draft for that...

File: backend/src/authentication/PlaintextAuthenticationProvider.ts (hunk 2 above)
Comment (srieger1) [reply backend/src/authentication/PlaintextAuthenticationProvider.ts]: I also added a constructor to the new provider to check existance of the required env var

**Pull request merged**